### PR TITLE
refactor: rename SchedulerCache to Cache in Scheduler

### LINKED ISSUE #108240: rename SchedulerCache to Cache in Scheduler
### What would you like to be added?

https://github.com/kubernetes/kubernetes/blob/f0d5ea1e1d4d43cd5caac92800207fcd2a62538c/pkg/scheduler/scheduler.go#L67-L70

We are doing some refactor works in https://github.com/kubernetes/kubernetes/pull/107135, and I found `SchedulerCache` seems redundant, would like to rename `SchedulerCache` to `Cache`

### Why is this needed?

`SchedulerCache` seems redundant as a field of `Scheduler`.

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Refactoring for idiomatic Go code

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/108240

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ahg-g** [comment]
can you also change [schedulerCache](https://github.com/kubernetes/kubernetes/blob/86945d7e74348045d1ae56b174748bc6b9d8887d/pkg/scheduler/internal/cache/cache.go#L56) to cacheImpl?

File: pkg/scheduler/internal/cache/cache.go [hunk 1]
```diff
@@ -90,15 +90,15 @@ type imageState struct {
...
+func (cache *cacheImpl) createImageStateSummary(state *imageState) *framework.ImageStateSummary {
 	return &framework.ImageStateSummary{
 		Size:     state.size,
 		NumNodes: len(state.nodes),
 	}
 }
 
-func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *schedulerCache {
-	return &schedulerCache{
+func newSchedulerCache(ttl, period time.Duration, stop <-chan struct{}) *cacheImpl {
```
Comment (denkensk) [inline pkg/scheduler/internal/cache/cache.go:100]: `newSchedulerCache` --> `newCacheImpl` or just `newCache`?

File: pkg/scheduler/eventhandlers_test.go [hunk 2]
```diff
@@ -228,18 +228,18 @@ func TestUpdatePodInCache(t *testing.T) {
 			schedulerCache := cache.New(ttl, ctx.Done())
 			schedulerQueue := queue.NewTestQueue(ctx, nil)
 			sched := &Scheduler{
-				SchedulerCache:  schedulerCache,
+				Cache:           schedulerCache,
 				SchedulingQueue: schedulerQueue,
```
Comment (denkensk) [inline pkg/scheduler/eventhandlers_test.go:232]:
```suggestion
				Cache:           cache.New(ttl, ctx.Done()),
				SchedulingQueue: queue.NewTestQueue(ctx, nil),
```

File: pkg/scheduler/internal/cache/cache.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/internal/cache/cache.go:100]: At first, I renamed this to `newCacheImpl`  , but then I realized it's just one implementation, so I changed back. `newCache` seems good to me.

File: pkg/scheduler/eventhandlers_test.go (hunk 2 above)
Comment (kerthcet) [reply pkg/scheduler/eventhandlers_test.go:232]: done

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by kerthcet** [comment]
As this pr promoted faster, I'd like to merge this first. cc @ruquanzhao , you may have to rebase your pr then.
/hold cancel

**Comment by denkensk** [comment]
Squash ?

**Comment by kerthcet** [comment]
> Squash ?

I'd like to keep two commits here as they are independent. Make sense?

**Comment by alculquicondor** [comment]
All good

/lgtm

**Comment by denkensk** [comment]
Approve is also needed. 😄

**Comment by alculquicondor** [comment]
I already approved. You are missing one from test owners

**Comment by aojea** [comment]
> I already approved. You are missing one from test owners

/approve

**Pull request merged**